x-pack/plugins/ingest_manager/server/types/models/agent_config.ts

```diff
@@ -9,7 +9,7 @@ import { AgentConfigStatus } from '../../../common';
 
 const AgentConfigBaseSchema = {
   name: schema.string(),
-  namespace: schema.maybe(schema.string()),
+  namespace: schema.string({ minLength: 1 }),
```

Comment:
@nchaulet Can we use this to also do some validation? For example namespace should not contain - as an example.

Reply:
Yes we can provide a validate function, where we can add all the custom logic we want